ssl/ssl_ciph.c

```diff
@@ -995,14 +995,8 @@ static int ssl_cipher_process_rulestr(const char *rule_str,
             ch = *l;
             buf = l;
             buflen = 0;
-#ifndef CHARSET_EBCDIC
-            while (((ch >= 'A') && (ch <= 'Z')) ||
-                   ((ch >= '0') && (ch <= '9')) ||
-                   ((ch >= 'a') && (ch <= 'z')) ||
-                   (ch == '-') || (ch == '.') || (ch == '='))
-#else
-            while (isalnum(ch) || (ch == '-') || (ch == '.') || (ch == '='))
-#endif
+            while (!(ch & ~127) && (isalnum(ch) || ch == '-' || ch == '.'
+                                    || ch == '='))
```

Comment:
These are the kinds of inconsistencies I've been discovering.
Some things look like they are being processed in the native character set but possibly aren't. Somethings look to mix the two modes and are definitely incorrect.

As far as I've been able to ascertain, characters are dealt with in the native character set -- thus loading certificates from file means the native char set.  There are a few exceptions where the alternative is used, these are mostly documented.  It very much seems like a kludge hacked onto the side to mostly get it working.